app/src/main/java/org/mozilla/reference/browser/settings/AboutFragment.kt

```diff
@@ -31,6 +32,26 @@ class AboutFragment : Fragment() {
         return inflater.inflate(R.layout.fragment_about, container, false)
     }
 
+    private fun getVersionFromExtensionManifest(path: String): String {
+        val stream = context?.assets?.open("$path/manifest.json")
+        val reader = JsonReader(stream?.bufferedReader())
+        reader.beginObject()
+        try {
+            while (reader.hasNext()) {
+                val name = reader.nextName()
+                if (name == "version") {
+                    return reader.nextString()
+                }
+                reader.skipValue()
+            }
+            reader.endObject()
+            return ""
+        } finally {
+            reader.close()
+            stream?.close()
+        }
+    }
```

Comment:
Cool! But, should it be not a `suspend` function? Doesn't it execute on the main thread?

Reply:
Yes, this is on main thread. What's the best way to use a suspend function in a Fragment? Which coroutine scope should be used and how to make `onViewCreated` async? Are there some examples of this pattern somewhere?

Reply:
I think that actually we can generate it at compile time and add it as a R.string. Let me check.

Reply:
Something like this: https://github.com/sammacbeth/daisy/pull/1